### PR TITLE
docs: generate typescript data models

### PR DESCRIPTION
**Pull request by md-saif-husain**
**Description**
This PR enables the user to generate Typescript data models
- added folder named as generate-typescript-models under examples 
- adapt [example overiew](https://github.com/md-saif-husain/modelina/blob/docs/generate-typescript-models(%23411)/examples/README.md) to list generate typescript data model
- adapt [usage.md](https://github.com/md-saif-husain/modelina/blob/docs/generate-typescript-models(%23411)/docs/usage.md#generate-typescript-models) to link example

**Related issue(s)**

Resolves[#411](https://github.com/asyncapi/modelina/issues/411)

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -50,7 +50,9 @@ TODO
 TODO 
 
 ## Generate TypeScript models
-TODO 
+There are one way to generate models from object.
+
+- [Generate Data Model](../examples/generate-typescript-models)
```
Comment (jonaslagoni) [inline docs/usage.md:55]: Maybe something along these lines is better:

```suggestion
Among many of the output languages we support, TypeScript is one of them. You can check out the [TypeScript documentation for more advanced use-cases](./languages/TypeScript.md).

Check out this [example out for a basic live demonstration](../examples/generate-typescript-models).
```

Thoughts @md-saif-husain ?

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
@md-saif-husain thanks for the PR! Just a small doc comment from my side 👍

File: docs/usage.md (hunk 1 above)
Comment (md-saif-husain) [reply docs/usage.md:55]: @jonaslagoni : Thanks for feedback . Above suggestion make sense . I will update the usage.md

Comment (md-saif-husain) [reply docs/usage.md]: @jonaslagoni : I have updated usage.md as per your suggestion

**Review by jonaslagoni — DISMISSED** [review]
LGTM 👍 

@alequetzalli anything from your side?

File: docs/usage.md [hunk 2]
```diff
@@ -50,7 +50,8 @@ TODO
 TODO 
 
 ## Generate TypeScript models
-TODO 
+Among many of the output languages we support, TypeScript is one of them. You can check out the [TypeScript documentation for more advanced use-cases](./languages/TypeScript.md).
+Check out this [example out for a basic live demonstration](../examples/generate-typescript-models).
```
Comment (alequetzalli) [inline docs/usage.md:54]:
```suggestion
TypeScript is one of the many output languages we support. Check out the following [TypeScript documentation for more advanced use-cases](./languages/TypeScript.md).

Or, take a look at this [basic example for a live demonstration](../examples/generate-typescript-models).
```

File: examples/README.md [hunk 3]
```diff
@@ -7,4 +7,5 @@ This directory contains a series of self-contained examples that you can use as
 - [asyncapi-from-parser](./asyncapi-from-parser) - A basic example where a AsyncAPI JS object from the [parser-js](https://github.com/asyncapi/parser-js) is used to generate models.
```
Comment (alequetzalli) [inline examples/README.md:7]:
```suggestion
- [asyncapi-from-parser](./asyncapi-from-parser) - A basic example where an AsyncAPI JS object from the [parser-js](https://github.com/asyncapi/parser-js) is used to generate models.
```

**Review by alequetzalli — COMMENTED** [review]
Okidoki, left a few tiny suggestions. 😀👍🏽

**Comment by jonaslagoni** [comment]
@all-contributors please add @md-saif-husain for example, test and doc

**Pull request merged**